mainwindow.py

```diff
@@ -1,1057 +1,1093 @@
+        """Remove an assigned a LCSC Part number to a footprint."""
+        for item in self.footprint_list.GetSelections():
+            row = self.footprint_list.ItemToRow(item)
+            ref = self.footprint_list.GetTextValue(row, 0)
+            fp = get_footprint_by_ref(GetBoard(), ref)[0]
+            self.store.set_lcsc(ref, "")
+        self.populate_footprint_list()
+
+    def select_alike(self, e):
+        """Select all parts that have the same value and footprint."""
+        num_sel = (
+            self.footprint_list.GetSelectedItemsCount()
+        )  # could have selected more than 1 item (by mistake?)
+        if num_sel == 1:
+            item = self.footprint_list.GetSelection()
+        else:
+            self.logger.warning(f"Select only one component, please.")
+            return
+        row = self.footprint_list.ItemToRow(item)
+        ref = self.footprint_list.GetValue(row, 0)
+        part = self.store.get_part(ref)
+        for r in range(self.footprint_list.GetItemCount()):
+            value = self.footprint_list.GetValue(r, 1)
+            fp = self.footprint_list.GetValue(r, 2)
+            if part[1] == value and part[2] == fp:
+                self.footprint_list.SelectRow(r)
+
+    def get_part_details(self, e):
+        """Fetch part details from LCSC and show them one after another each in a modal."""
+        parts = self.get_selected_part_id_from_gui()
+        if not parts:
+            return
+
+        for part in parts:
+            self.show_part_details_dialog(part)
+
+    def get_column_by_name(self, column_title_to_find):
+        """Lookup a column in our main footprint table by matching its title"""
+        for col in self.footprint_list.Columns:
+            if col.Title == column_title_to_find:
+                return col
+        return None
+
+    def get_column_position_by_name(self, column_title_to_find):
+        """Lookup the index of a column in our main footprint table by matching its title"""
+        col = self.get_column_by_name(column_title_to_find)
+        if not col:
+            return -1
+        return self.footprint_list.GetColumnPosition(col)
+
+    def get_selected_part_id_from_gui(self):
+        """Get a list of LCSC part#s currently selected"""
+        lcsc_ids_selected = []
+        for item in self.footprint_list.GetSelections():
+            row = self.footprint_list.ItemToRow(item)
+            if row == -1:
+                continue
+
+            lcsc_id = self.get_row_item_in_column(row, "LCSC")
+            lcsc_ids_selected.append(lcsc_id)
+
+        return lcsc_ids_selected
+
+    def get_row_item_in_column(self, row, column_title):
+        return self.footprint_list.GetTextValue(row, self.get_column_position_by_name(column_title))
+
+    def show_part_details_dialog(self, part):
+        wx.BeginBusyCursor()
+        try:
+            self.logger.info(f"Opening PartDetailsDialog window for part with value: '{part} (this should be "
```

Comment:
Is this a dev artifact or is there a reason why that should be logged whenever the part details dialog is opened?

Reply:
Probably a bit of a dev artifact at this point. I think I started this trying to just log what was causing the issue, then ended up doing the rest of this. It could be commented out or removed now, sure.  Was vaguely thinking if there were problems in the future or with http retrieval, this could provide a breadcrumb for users to either fix a busted part# or to file a bug report where the log would contain more useful info right before stuff went wrong 

I'm good either way, if you think it's too spammy let's remove it

Reply:
I think I'll comment it out